@here/harp-mapview/lib/text/TextStyleCache.ts

```diff
@@ -441,19 +459,123 @@ export class TextStyleCache {
-                    style.vAlignment === "Below"
-                        ? VerticalAlignment[style.vAlignment]
-                        : VerticalAlignment.Center,
-                horizontalAlignment:
-                    style.hAlignment === "Left" ||
-                    style.hAlignment === "Center" ||
-                    style.hAlignment === "Right"
-                        ? HorizontalAlignment[style.hAlignment]
-                        : HorizontalAlignment.Center
+                verticalAlignment,
+                horizontalAlignment,
+                placements
             }
         };
     }
 }
+
+function parseAlignmentAndPlacements(
+    hAlignment: string | null | undefined,
+    vAlignment: string | null | undefined,
+    placementsTokens: string | null | undefined
+): {
+    horizontalAlignment: HorizontalAlignment;
+    verticalAlignment: VerticalAlignment;
+    placements: TextPlacements;
+} {
+    // Currently supported only for PoiTechnique.
+    const placements: TextPlacements = placementsTokens
+        ? parseTechniquePlacements(placementsTokens)
+        : [];
+
+    // Ignore alignment attributes when placements attributes are defined or provide default
+    // values if none of them are defined.
+    // NOTE: Alignment ignore may be removed if we decide to support both attributes separately.
+    const horizontalAlignment =
+        placements.length > 0
+            ? hAlignFromPlacement(placements[0].h)
+            : parseTechniqueHAlignValue(hAlignment);
+
+    const verticalAlignment =
+        placements.length > 0
+            ? vAlignFromPlacement(placements[0].v)
+            : parseTechniqueVAlignValue(vAlignment);
+
+    return { horizontalAlignment, verticalAlignment, placements };
+}
+
+function parseTechniqueHAlignValue(hAlignment: string | undefined | null): HorizontalAlignment {
+    return hAlignment === "Left" || hAlignment === "Center" || hAlignment === "Right"
+        ? HorizontalAlignment[hAlignment]
+        : defaultTextLayoutStyle.horizontalAlignment;
+}
+
+function parseTechniqueVAlignValue(vAlignment: string | undefined | null): VerticalAlignment {
+    return vAlignment === "Above" || vAlignment === "Center" || vAlignment === "Below"
+        ? VerticalAlignment[vAlignment]
+        : defaultTextLayoutStyle.verticalAlignment;
+}
+
+function parseTechniquePlacements(placementsString: string | undefined | null): TextPlacements {
+    // Parse placement properties if available.
+    const placements: TextPlacements = [];
+    const placementsTokens = placementsString
+        ? placementsString!
+              .toUpperCase()
+              .replace(" ", "")
+              .split(",")
+        : [];
+    placementsTokens.forEach(p => {
+        const val = parseTechniquePlacementValue(p);
+        if (val !== undefined) {
+            placements.push(val);
+        }
+    });
+    return placements;
+}
+
+function parseTechniquePlacementValue(p: string): TextPlacement | undefined {
```

Comment:
it would be good to cover function with a unit test, either testing the fuction directly or indirectly through a unit test of the TextStyleCache class.